bn_fast_mp_invmod.c

```diff
@@ -53,7 +53,7 @@ int fast_mp_invmod(const mp_int *a, const mp_int *b, mp_int *c)
    if ((res = mp_copy(&y, &v)) != MP_OKAY) {
       goto LBL_ERR;
    }
-   mp_set(&D, 1);
+   mp_set(&D, (mp_digit)1);
```

Comment:
can you please explain what's the purpose of this cast?

this was introduced in "cast mp_digit in call function" (currently 077d5891ad1b0c5c7314fd964b50061824952d66)

shouldn't this cast be implicit!?

Reply:
Yes, currently, the cast is implicit.
The C language has many implicit behaviours, and it is a big issue of C.
With [MISRA](https://en.wikipedia.org/wiki/MISRA_C) rules, all behaviour must be explicit.
The source becomes optimized for human code review (and the compiler generates the same thing).

Reply:
ew, I already thought there will be something like MISRA in your reply... if that's the only reason for the uglification I'd vote against this patch
also with the background that I don't know a single person who writes code like that (if he doesn't have to), so all changes that come in won't be in that format and you'd always have to fix it up ;)
@tomstdenis your opinion? :)